runtime/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSessionFactory.java

```diff
@@ -614,11 +583,57 @@ private void onMqttData(
 
                     kafkaPayload = payload.sizeof() > 0 ? mqttSessionStateRO.wrap(buffer, offset, limit) : EMPTY_OCTETS;
 
-                    session.doKafkaData(traceId, authorization, budgetId,
-                        reserved, flags, kafkaPayload, kafkaDataEx);
+                    session.doKafkaData(traceId, authorization, budgetId, reserved,
+                        sessionPadding, flags, kafkaPayload, kafkaDataEx);
                     break;
                 }
```

Comment:
Same feedback here for this switch across session will and state - please refactor to separate methods.